godo.go

```diff
@@ -223,7 +223,9 @@ func (c *Client) Do(req *http.Request, v interface{}) (*Response, error) {
 		return nil, err
 	}
 
-	defer resp.Body.Close()
+	defer func() {
+		err = resp.Body.Close()
```

Comment:
what i should do is check for the err first in this case. i'll send some updated code.